app/services/arcgis_api/geocoder.rb

```diff
@@ -49,9 +49,9 @@ def request_headers
 
     def parse_suggestions(response_body)
       if response_body['error']
-        if response_body['error']['code'] >= 400 && response_body['error']['code'] < 500
-          raise Faraday::ClientError.new(response_body)
-        end
+        error_code = response_body.dig('error', 'code')
+
+        raise Faraday::ClientError.new(RuntimeError.new("received error code #{error_code}"), response_body)
```

Comment:
Could we add / update specs to clarify what we're expecting with these changes? It'd also help from a review perspective to understand the intended effect.

Something like this?

```rb
    it 'returns an error response body but with Status coded as 200' do
      stub_request_suggestions_error

      expect { subject.suggest('100 Main') }.to raise_error do |error|
        expect(error).to be_instance_of(Faraday::ClientError)
        expect(error.message).to eq('received error code 400')
        expect(error.response).to be_kind_of(Hash)
      end
    end
```

Reply:
+1 to adding specs